src/megaapi_impl.cpp

```diff
@@ -8236,6 +8244,85 @@ void MegaApiImpl::getversion_result(int versionCode, const char *versionString,
     fireOnRequestFinish(request, MegaError(e));
 }
 
+void MegaApiImpl::getlocalsslcertificate_result(m_time_t ts, string *certdata, error e)
+{
+    MegaError megaError(e);
+    if(requestMap.find(client->restag) == requestMap.end()) return;
+    MegaRequestPrivate* request = requestMap.at(client->restag);
+    if(!request || (request->getType() != MegaRequest::TYPE_GET_LOCAL_SSL_CERT)) return;
+
+    if (!e)
+    {
+        string result;
+        const char *data = certdata->data();
+        const char *enddata = certdata->data() + certdata->size();
+        MegaStringMapPrivate *datamap = new MegaStringMapPrivate();
+        for (int i = 0; data < enddata; i++)
+        {
+            result = i ? "-----BEGIN CERTIFICATE-----\n"
+                       : "-----BEGIN PRIVATE KEY-----\n";
+
+            const char *end = strstr(data, ";");
+            if (!end)
+            {
+                if (!i)
+                {
+                    delete datamap;
+                    fireOnRequestFinish(request, MegaError(API_EINTERNAL));
+                    return;
+                }
+                end = enddata;
+            }
+
+            while (data < end)
+            {
+                int remaining = end - data;
+                if (remaining > 64)
+                {
+                    result.append(data, 64);
+                    data += 64;
+                }
+                else
+                {
+                    result.append(data, remaining);
+                    data += remaining;
+                }
```

Comment:
I would have written the blocks above like:
```
int dataSize = ((end - data) > 64) ? 64 : (end - data);
result.append(data, dataSize);
data += dataSize;
```
But your code maybe it's clearer.